File: Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        
        max_current=max_global=nums[0]
        for i in range(1, n):
            max_current = max(nums[i], max_current+nums[i])
            max_global = max(max_global, max_current)
        return max_global
        
        

File: solutions/Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int global_max = nums[0];
        int current_max = nums[0];
        int n = nums.size();
        for (int i=1; i<n; i++){
            current_max = max(nums[i], current_max+nums[i]);
            global_max = max(current_max, global_max);
        }
        return global_max;
        
    }
};